scheduler/task.go

```diff
@@ -537,6 +535,37 @@ func (t *task) fire() {
 	t.state = core.TaskSpinning
 }
 
+// disable proceeds disabling a task which consists of changing task state to disabled, emitting an appropriate event
+// and unsubscribing its dependencies
+func (t *task) disable(failureMsg string) {
+	t.Lock()
+	t.state = core.TaskDisabled
+	t.Unlock()
+
+	// Send task disabled event
+	event := new(scheduler_event.TaskDisabledEvent)
+	event.TaskID = t.id
+	event.Why = fmt.Sprintf("Task disabled with error: %s", failureMsg)
+	defer t.eventEmitter.Emit(event)
+
+	// We need to unsubscribe from deps when a task has disabled
+	t.UnsubscribePlugins()
+}
+
+// end proceeds ending a task which consists of changing task state to ended, emitting an appropriate event
+// and unsubscribing its dependencies
+func (t *task) end() {
+	t.Lock()
+	t.state = core.TaskEnded
+	t.Unlock()
+	event := new(scheduler_event.TaskEndedEvent)
+	event.TaskID = t.id
+	defer t.eventEmitter.Emit(event)
+
+	// We need to unsubscribe from deps when a task has ended
+	t.UnsubscribePlugins()
```

Comment:
do we allow ended task be restarted? If the answer is yes, do we subscribe plugins in the restart?